packages/ui/src/components/form-primitives/form-wrapper.tsx

```diff
@@ -1,27 +1,41 @@
 import { FormHTMLAttributes, ReactNode } from 'react'
+import { FieldValues, FormProvider, UseFormReturn } from 'react-hook-form'
 
 import { cn } from '@utils/cn'
 
-interface FormWrapperProps extends FormHTMLAttributes<HTMLFormElement> {
+interface FormWrapperProps<T extends FieldValues> extends FormHTMLAttributes<HTMLFormElement>, UseFormReturn<T> {
   className?: string
-  formRef?: (form: HTMLFormElement) => void
+  formRef?: React.Ref<HTMLFormElement> | null
```

Comment:
why do we need to specify `null` as a valid type?

Reply:
I added for `ForwardType` and checked earlier and for that it was needed. `React.Ref` type can handle it.